multiqc/modules/damageprofiler/damageprofiler.py

```diff
@@ -0,0 +1,238 @@
+            log.warn("Could not parse DamageProfiler JSON: '{}'".format(f['fn']))
+            return None
+        
+        #Get sample name from JSON first
+        s_name = parsed_json['metadata']['sample_name']
+        self.add_data_source(f, s_name)
+
+        
+        #Add 3' G to A data 
+        self.threepGtoAfreq_data[s_name] = parsed_json['dmg_3p']
+
+        #Add 5' C to T data
+        self.fivepCtoTfreq_data[s_name] = parsed_json['dmg_5p']
+
+        #Add lendist forward 
+        self.lgdist_fw_data[s_name] = parsed_json['lendist_fw']
+
+        #Add lendist reverse
+        self.lgdist_rv_data[s_name] = parsed_json['lendist_rv']
+
+    
+    #### Tables from here on 
+
+    def dmgprof_misinc_stats(self, dict_to_plot, readend, substitution):
+        """ Take the parsed stats from the DamageProfiler and add it to the
+        basic stats table at the top of the report """
+
+        headers = OrderedDict()
+        headers['1'] = {
+            'title': readend + ' ' + substitution + ' 1st base ',
+            'description': readend + ' ' + substitution + ' 1st base ',
+            'min': 0,
+            'max': 100,
+            'suffix': '%',
+            'scale': 'GrRd',
+            'format': '{:,.2f}',
+            'modify': lambda x: x * 100.0
+        }
+        headers['2'] = {
+            'title': readend + ' ' + substitution + ' 2nd base ',
+            'description': readend + ' ' + substitution + ' 2nd base ',
+            'min': 0,
+            'max': 100,
+            'suffix': '%',
+            'scale': 'GrRd',
+            'format': '{:,.2f}',
+            'modify': lambda x: x * 100.0
+        }
+
+        # Create new small subset dictionary for entries (we need just the first two data (k,v) pairs from each report)
+        data = OrderedDict()
+        dict_to_add = dict()
+        
+        for key in dict_to_plot.keys():
+            tmp = dict_to_plot[key]
+            pos = list(range(1,3))
+            strlist = list(map(float,tmp[:2]))
+            tuples = list(zip(str(pos),strlist))
+            data = dict((x, y) for x, y in tuples)
+            #Extract first two elements from list
+            dict_to_add[key] = data
+        pprint.pprint(dict_to_add)
+        
+        self.general_stats_addcols(dict_to_add,headers)
+    
+
+     
+
+    #### Plotting from here on
+    #Nice Linegraph plot for lgdist data
+
+    def lgdistplot(self,dict_to_use,orientation):
+        """Generate a read length distribution plot"""
+
+        data = dict()
+        for s_name in dict_to_use:
+            try:
+                data[s_name] = {int(d): int (dict_to_use[s_name][d]) for d in dict_to_use[s_name]}
+            except KeyError:
+                pass
+        if len(data) == 0:
+            log.debug('No valid data for forward read lgdist input!')
+            return None
+        
+        config = {
+            'id': 'length-distribution-' + orientation,
+            'smooth_points': 50,
```

Comment:
This seems kind of overkill? I usually smooth down to 500 - 1000 if there can be many thousands of points. If this is a read length plot for illumina data then no need to smooth I think.